docs/berkeley-upgrade/migrating-archive-database-to-berkeley.mdx

```diff
@@ -13,225 +13,419 @@ keywords:
+   --migrated-archive-uri {migrated_connection_string}  
 ```
 
-#### Stage 2: Incremental migration
+where:
+
+ - `mainnet-archive-uri` - connection string to the original Devnet/Mainnet archive database 
+ - `migrated-archive-uri` - connection string tothe migrated archive database
+
+ Note: When migrating devnet archive database, pass the connection string of the Devnet database to the flag `--mainnet-archive-uri`. 
+ 
+### Stage 2: Incremental migration
 
-After the initial migration, the data is migrated data up to the last canonical block. However, `mainnet` data is progressing with new blocks that must also be migrated again and again until the fork block is announced. 
+After the initial migration, the data is migrated data up to the last canonical block. However, Devnet/Mainnet data is progressing with new blocks that must also be migrated again and again until the fork block is announced. 
 
+:info: Incremental migration can, and probably must, be repeated a couple of times until the fork block is announced by Mina Foundation.
+Run the incremental migration multiple times with the latest Devnet/Mainnet database and the latest replayer checkpoint file.
 
 - Inputs
-   - Newest mainnet database
-   - Mainnet genesis ledger
+   - Latest Devnet/Mainnet database
+   - Devnet/Mainnet genesis ledger
    - Replayer checkpoint from last run 
-   - Migrated berkeley schema from initial migration 
+   - Migrated berkeley database from initial migration 
 
 - Outputs
-   - Migrated mainnet database to berkeley from up to last canonical block
-   - Replayer checkpoint which can be used for next incremental migration
+   - Migrated Devnet/Mainnet database to the Berkeley format up to the last canonical block
+   - Replayer checkpoint which can be used for the next incremental migration
 
-##### Phase 1: Berkeley migration app run
+### Phase 1: Berkeley migration app run
 
 ```
-mina-berkeley-migration --batch-size 1000 --config-file genesis_ledgers/mainnet.json
---mainnet-archive-uri {mainnet_connection_string} --migrated-archive-uri {migrated_connection_string} --mainnet-blocks-bucket 
-{bucket name} --network "mainnet"
+mina-berkeley-migration \
+   --batch-size 1000 \
+   --config-file FILE \
+   --mainnet-archive-uri {mainnet_connection_string} \
+   --migrated-archive-uri {migrated_connection_string} \
+   --blocks-bucket {bucket name} \
+   --keep-precomputed-blocks \
+   --stream-precomputed-blocks \
+   --network NETWORK
 ```
-where: 
-`--network` - precomputed blocks prefix 
 
+where:
+
+`--network` - the network name (`devnet` or `mainnet`) when determining precomputed blocks. Precomputed blocks are assumed to be named with format: `{network}-{height}-{state_hash}.json`
+
+`--config-file` - Use genesis_ledgers/devnet.json for Devnet migration and genesis_ledgers/mainnet.json for Mainnet migration 
+
+Note: When migrating the Devnet archive database, pass the connection string of the devnet database to the flag `--mainnet-archive-uri`.
```

Comment:
other sections define all of the inputs, here we are inconsistent and just define a few